### ISSUE TITLE
[BUG] with strict-ssl set to false v10 now does bad requests and gets "the plain http request was sent to HTTPS port"

### ISSUE THREAD

**Issue by lukeapage** [npm/cli#6948]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### This issue exists in the latest npm version

- [X] I am using the latest npm

### Current Behavior

We set strict-ssl to false because we have a corporate proxy on our own certificate and not setting it we get an error

```
npm ERR! code UNABLE_TO_VERIFY_LEAF_SIGNATURE
npm ERR! errno UNABLE_TO_VERIFY_LEAF_SIGNATURE
npm ERR! request to https://registry.npmjs.org/write-pkg/-/write-pkg-4.0.0.tgz failed, reason: unable to verify the first certificate
```

In v9 of npm it installs ok.

In v10 we get error requests from npmjs:
```
<html>
<head><title>400 The plain HTTP request was sent to HTTPS port</title></head>
<body>
<center><h1>400 Bad Request</h1></center>
<center>The plain HTTP request was sent to HTTPS port</center>
<hr><center>cloudflare</center>
</body>
</html>
```

### Expected Behavior

I expect no errors like in v9.

### Steps To Reproduce

In windows (not sure if its important)
set strict-ssl to false
run npm install

I am assuming it doesn't make a difference, but I have a proxy setting:
proxy=http://localhost:8888/
which goes to fiddler, which proxies the requests

### Environment

- npm: v10.2.0
- Node.js: v21.1.0
- OS Name: Windows 10
- npm config:
```ini
prefix=c:\npmprefix
cache=c:\npmcache
strict-ssl=false
proxy=http://localhost:8888/
```

**Comment by wraithgar** [comment]
It looks like your proxy isn't using https when it forwards the request to the npm registry. There's not much npm itself can do about that.

**Comment by lukeapage** [comment]
How come it only fails with npm 10 and only when strict ssl is false ? That shouldn’t change the behavior of outgoing requests..

**Comment by lukeapage** [comment]
Failed request - raw view of the request coming from npm v9:

![image](https://github.com/npm/cli/assets/309321/cac075ff-84b2-487b-80b6-26ac0d008968)

Notice the protocol is http

then the only change is upgrading npm to v10

![image](https://github.com/npm/cli/assets/309321/5b1831d3-70db-4a3c-8fb5-a5a86d114a4f)

This is within fiddler on the machine I am running npm - so it is the "first hop" before leaving the machine.

@wraithgar please can you re-open this bug.

**Comment by aivanise** [comment]
I also have a http:// registry but npm 11.0.0 is working ok, when trying to upgrade to 11.4.2 i see exact same behaviour as you, my registry URL gets silently rewritten to https

strict-ssl=false makes no difference, haven't found any other ssl related config

```
[root@builder-alex-nodejs btife]# cat client/.npmrc

registry=http://repo.2e-systems.com:8081/nexus/repository/npm_group/
strict-ssl=false

[root@builder-alex-nodejs btife]# npm --verbose ci --prefix ./client |& fgrep SSL

npm http fetch GET https://repo.2e-systems.com:8081/nexus/repository/npm_group/zoom-level/-/zoom-level-2.5.0.tgz attempt 1 failed with ERR_SSL_WRONG_VERSION_NUMBER
npm http fetch GET https://repo.2e-systems.com:8081/nexus/repository/npm_group/yup/-/yup-1.6.1.tgz attempt 1 failed with ERR_SSL_WRONG_VERSION_NUMBER
npm http fetch GET https://repo.2e-systems.com:8081/nexus/repository/npm_group/yocto-queue/-/yocto-queue-0.1.0.tgz attempt 1 failed with ERR_SSL_WRONG_VERSION_NUMBER
npm http fetch GET https://repo.2e-systems.com:8081/nexus/repository/npm_group/yaml/-/yaml-1.10.2.tgz attempt 1 failed with ERR_SSL_WRONG_VERSION_NUMBER
npm http fetch GET https://repo.2e-systems.com:8081/nexus/repository/npm_group/yallist/-/yallist-3.1.1.tgz attempt 1 failed with ERR_SSL_WRONG_VERSION_NUMBER
npm http fetch GET https://repo.2e-systems.com:8081/nexus/repository/npm_group/yargs-parser/-/yargs-parser-21.1.1.tgz attempt 1 failed with ERR_SSL_WRONG_VERSION_NUMBER`
```

**Comment by aivanise** [comment]
seems to be fixed in npm 11.6.2

not even strict-ssl=false seems to be necessary in .npmrc
